### PR TITLE
doc: make explanation easier to undrstand

### PR DESCRIPTION
**Pull request by tshepang**
That sentence make me read it a few times before properly understanding it

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -791,8 +791,7 @@ trait Foo {
     fn bar(&self);
 }
 
-// we now declare a function which takes an object with Foo trait implemented
-// as parameter
+// we now declare a function which takes an object implementing the Foo trait
```
Comment (Gankro) [inline src/librustc/diagnostics.rs]: "that implements the Foo trait" is probably the right... tense..?

Comment (tshepang) [inline src/librustc/diagnostics.rs]: I think this is correct, and it sounds better anyways. Having *which* and *that* in a sentence sounds awkward.

Comment (Gankro) [inline src/librustc/diagnostics.rs]: I don't really agree; but I do agree that this is unambiguously an improvement over the old wording.

**Comment by Gankro** [comment]
@bor r+ rollup

**Comment by Gankro** [comment]
:bow:

**Pull request merged**